src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +42,1136 @@ const ActorInit En_Niw_InitVars = {
+                this->unk_296++;
+                this->timer5 = 10;
+            } else {
+                osSyncPrintf("\n\n");
+                osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 発生できず  ☆☆☆☆☆ \n" VT_RST);
+            }
+        }
+    }
+}
+
+void func_80AB6100(EnNiw* this, GlobalContext* globalCtx, s32 arg2) {
+    f32 factor;
+    f32 targetRotY;
+
+    if (this->timer4 == 0) {
+        this->timer4 = 3;
+
+        if (this->actor.bgCheckFlags & 1) {
+            this->actor.velocity.y = 3.5f;
+        }
+    }
+    if (this->timer3 == 0) {
+        this->unk_2A0++;
+        this->unk_2A0 &= 1;
+        this->timer3 = 5;
+    }
+    if (this->unk_2A0 == 0) {
+        factor = D_80AB860C[arg2];
+    } else {
+        factor = -D_80AB860C[arg2];
+    }
+    if (arg2 == 1) {
+        if (this->timer6 == 0 || this->actor.bgCheckFlags & 8) {
+            this->timer6 = 150;
+            if (this->timer8 == 0) {
+                this->timer8 = 70;
+                this->unk_2E4 = this->actor.yawTowardsLink;
+            }
+        }
+    }
+    targetRotY = this->unk_2E4 + factor;
+    Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, targetRotY, 3, this->unk_2FC, 0);
+    Math_SmoothScaleMaxF(&this->unk_2FC, 3000.0f, 1.0f, 500.0f);
+    func_80AB5BF8(this, globalCtx, 5);
+}
+
+void EnNiw_ResetAction(EnNiw* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060000E8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060000E8), 0, -10.0f);
+
+    switch (this->actor.params) {
+        case 0x4:
+            this->actionFunc = func_80AB6450;
+            break;
+        case 0xD:
+            this->actionFunc = func_80AB6324;
+            break;
+        default:
+            this->actionFunc = func_80AB6570;
+            break;
+    }
+}
+
+void func_80AB6324(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->unk_308 != 0) {
+        this->actor.velocity.y = Math_Rand_ZeroFloat(2.0f) + 4.0f;
+        this->actor.speedXZ = Math_Rand_ZeroFloat(2.0f) + 3.0f;
+        this->actionFunc = func_80AB63A8;
+    }
+    func_80AB5BF8(this, globalCtx, 1);
+}
+
+void func_80AB63A8(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1 && this->actor.velocity.y < 0.0f) {
+        this->unk_2AC = this->unk_2B8 = this->actor.posRot.pos.x;
+        this->unk_2B0 = this->unk_2BC = this->actor.posRot.pos.y;
+        this->unk_2B4 = this->unk_2C0 = this->actor.posRot.pos.z;
+        this->timer5 = this->timer4 = this->unk_29E = 0;
+
+        this->unk_26C[7] = this->unk_26C[5] = this->unk_26C[6] = this->unk_26C[8] = this->actor.speedXZ =
+            this->unk_2FC = this->unk_300 = 0.0f;
+
+        this->actionFunc = func_80AB6570;
+    } else {
+        func_80AB5BF8(this, globalCtx, 2);
+    }
+}
+
+void func_80AB6450(EnNiw* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->actor.xzDistFromLink < 30.0f && fabsf(this->actor.posRot.pos.y - player->actor.posRot.pos.y) < 5.0f) {
```

Comment:
I'd like to see what others think, but this function may look better as an `if, else if, else` instead of the two returns